packages/calling-component-bindings/src/videoGallerySelector.ts

```diff
@@ -129,31 +129,29 @@ const sortedRemoteParticipants = (
     }
   });
 
-  const speakersList: Record<string, number> = {};
-  dominantSpeakers?.forEach((speaker, idx) => (speakersList[speaker] = idx));
-
   // If dominantSpeakers are available, we sort the video tiles basis on dominant speakers.
   if (dominantSpeakers) {
     participantsWithVideo.sort((a, b) => {
-      const idxA = speakersList[a.userId];
-      const idxB = speakersList[b.userId];
+      const idxA = dominantSpeakers[a.userId];
+      const idxB = dominantSpeakers[b.userId];
       if (idxA === undefined && idxB === undefined) return 0; // Both a and b don't exist in dominant speakers.
       if (idxA === undefined && idxB >= 0) return 1; // b exists in dominant speakers.
       if (idxB === undefined && idxA >= 0) return -1; // a exists in dominant speakers.
       return idxA - idxB;
     });
 
     participantsWithoutVideo.sort((a, b) => {
-      const idxA = speakersList[a.userId];
-      const idxB = speakersList[b.userId];
+      const idxA = dominantSpeakers[a.userId];
+      const idxB = dominantSpeakers[b.userId];
       if (idxA === undefined && idxB === undefined) return 0; // Both a and b don't exist in dominant speakers.
       if (idxA === undefined && idxB >= 0) return 1; // b exists in dominant speakers.
       if (idxB === undefined && idxA >= 0) return -1; // a exists in dominant speakers.
       return idxA - idxB;
     });
   }
 
-  return participantsWithVideo.concat(participantsWithoutVideo);
```

Comment:
nitty nit: Earlier direct return was better?

Reply:
All this code will be removed in a subsequent PR. ;)